src/libopensc/ctx.c

```diff
@@ -860,9 +860,18 @@ int sc_get_cache_dir(sc_context_t *ctx, char *buf, size_t bufsize)
 {
 	char *homedir;
 	const char *cache_dir;
+        scconf_block *conf_block = NULL;
 #ifdef _WIN32
 	char temp_path[PATH_MAX];
 #endif
+        conf_block = sc_get_conf_block(ctx, "framework", "pkcs15", 1);
+	cache_dir = scconf_get_str(conf_block, "file_cache_dir", NULL);
```

Comment:
On Thu, 05 Feb 2015 18:15:26 -0800
Frank Morgner <notifications@github.com> wrote:

> >  #ifdef _WIN32
> >  	char temp_path[PATH_MAX];
> >  #endif
> > +        conf_block = sc_get_conf_block(ctx, "framework", "pkcs15", 1);
> > +	cache_dir = scconf_get_str(conf_block, "file_cache_dir", NULL);
> 
> You should provide a reasonable default here. Presumably the caching-directory we used to have...
> 

if the caching-directory is not found, the old one will be used
there is.
  if (cache_dir != NULL) {

The reason why I did not put it into the scconf_get_str() line was:
The default one is not a string constant. It is constructed. So why the
default path should be constructed if the result is not used.